apps/src/labs/lab2.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import {levelsForLessonId} from '@cdo/apps/code-studio/progressReduxSelectors';
+import {ProgressState} from '@cdo/apps/code-studio/progressRedux';
+import {LevelWithProgress} from '@cdo/apps/types/progressTypes';
+import {getStandaloneProjectId} from '@cdo/apps/labs/projects/utils';
+import ProjectContainer from '@cdo/apps/labs/projects/ProjectContainer';
+import ProgressContainer from '@cdo/apps/labs/progress/ProgressContainer';
+import StandaloneVideo2 from '@cdo/apps/standaloneVideo2/StandaloneVideo2';
+import MusicView from '@cdo/apps/music/views/MusicView';
+import {useSelector} from 'react-redux';
+
+const Lab2: React.FunctionComponent = () => {
+  const currentApp = useSelector(
+    (state: {progress: ProgressState}) =>
+      levelsForLessonId(state.progress, state.progress.currentLessonId).find(
+        (level: LevelWithProgress) => level.isCurrentLevel
+      ).app
+  );
+
+  const channelId: string | null =
+    currentApp === 'music' ? getStandaloneProjectId() : null;
+
+  return (
+    <ProjectContainer channelId={channelId || undefined}>
```

Comment:
`getStandaloneProjectId` returns `string | null` but `ProjectContainer` wants `string | undefined`.  Should we make these consistent?

Reply:
Yeah, `string | undefined` should be fine, especially after Molly's change [here](https://github.com/code-dot-org/code-dot-org/pull/52523) since we're no longer explicitly returning null.

Reply:
Great; I'll update the return value type of `getStandaloneProjectId` and tidy this up.